posthog/api/insight.py

```diff
@@ -79,10 +90,25 @@
 from posthog.settings import CAPTURE_TIME_TO_SEE_DATA, SITE_URL
 from posthog.settings.data_stores import CLICKHOUSE_CLUSTER
 from posthog.user_permissions import UserPermissionsSerializerMixin
-from posthog.utils import DEFAULT_DATE_FROM_DAYS, relative_date_parse, should_refresh, str_to_bool
+from posthog.utils import DEFAULT_DATE_FROM_DAYS, refresh_requested_by_client, relative_date_parse, str_to_bool
 
 logger = structlog.get_logger(__name__)
 
+# default minimum wait time for refreshing an insight
+DEFAULT_INSIGHT_REFRESH_FREQUENCY = timedelta(minutes=15)
+
+# returns should_refresh, refresh_frequency
+def should_refresh_insight(insight: Insight) -> Tuple[bool, timedelta]:
+    refresh_frequency = DEFAULT_INSIGHT_REFRESH_FREQUENCY
+
+    if "interval" in insight.filters and insight.filters["interval"] == "hour":
+        refresh_frequency = timedelta(minutes=3)
+
+    if not insight.last_refresh:
```

Comment:
This is no more populated.

Reply:
It does actually still get populated

Reply:
Let's discuss - we should not be relying on it, it's a deprecated field. Note that `insight` model is also incorrect here since the insight might be from a dashboard - last_refresh is different on that.

Reply:
Did the refactor - makes sense